pkg/operations/kubernetesupgrade/upgradeagentnode.go

```diff
@@ -87,11 +87,19 @@ func (kan *UpgradeAgentNode) CreateNode(poolName string, agentNo int) error {
 		kan.ParametersMap,
 		nil)
 
-	if err != nil {
-		return err
+	if err == nil {
+		return nil
 	}
 
-	return nil
+	kan.logger.Errorf("Deployment %s failed with error %v", deploymentName, err)
+	// Get deployment error details
+	errResp, e := armhelpers.GetDeploymentError(kan.Client, kan.logger, kan.ResourceGroup, deploymentName)
```

Comment:
if I remember right there are two types of errors here. One is a problem during the deployment and the other is during validation. This assumes all are the former. To figure out which kind it is we should need to look at the response etc as well as the error.